tests/src/operator.rs

```diff
@@ -119,4 +125,180 @@ mod operator {
+            .await;
+
+        let target =
+            get_instance_name::<Deployment>(client.clone(), &service.name, &service.namespace)
+                .await
+                .unwrap();
+
+        let mut client_b = application
+            .run(
+                &format!("deployment/{target}"),
+                Some(&service.namespace),
+                Some(flags.clone()),
+                None,
+            )
+            .await;
+
+        client_b
+            .wait_for_line(Duration::from_secs(40), "Someone else is stealing traffic")
+            .await;
+
+        // check if client_a is stealing
+        let url = get_service_url(client, &service).await;
+        let client = reqwest::Client::new();
+        let req_builder = client.get(&url);
+        let mut headers = HeaderMap::default();
+        headers.insert("x-filter", "yes".parse().unwrap());
+        send_request(req_builder, Some("GET"), headers.clone()).await;
+
+        client_a.child.kill().await.unwrap();
+
+        let res = client_b.child.wait().await.unwrap();
+        assert!(!res.success());
+    }
+
+    #[cfg_attr(not(feature = "operator"), ignore)]
+    #[rstest]
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    pub async fn two_clients_steal_with_http_filter(
+        config_dir: &std::path::PathBuf,
+        #[future] service: KubeService,
+        #[future] kube_client: Client,
+        #[values(Application::NodeHTTP)] application: Application,
+    ) {
+        let service = service.await;
+        let kube_client = kube_client.await;
+        let url = get_service_url(kube_client.clone(), &service).await;
+
+        let mut config_path = config_dir.clone();
+        config_path.push("http_filter_header.json");
+
+        let flags = vec!["--steal"];
+
+        let mut client_a = application
+            .run(
+                &service.target,
+                Some(&service.namespace),
+                Some(flags.clone()),
+                Some(vec![("MIRRORD_CONFIG_FILE", config_path.to_str().unwrap())]),
+            )
+            .await;
+
+        client_a
+            .wait_for_line(Duration::from_secs(40), "daemon subscribed")
+            .await;
+
+        let mut config_path = config_dir.clone();
+        config_path.push("http_filter_header_no.json");
+
+        let mut client_b = application
+            .run(
+                &service.target,
+                Some(&service.namespace),
+                Some(flags),
+                Some(vec![("MIRRORD_CONFIG_FILE", config_path.to_str().unwrap())]),
+            )
+            .await;
+
+        client_b
+            .wait_for_line(Duration::from_secs(40), "daemon subscribed")
+            .await;
+
+        let client = reqwest::Client::new();
+        let req_builder = client.delete(&url);
+        let mut headers = HeaderMap::default();
+        headers.insert("x-filter", "yes".parse().unwrap());
+
+        send_request(req_builder, Some("DELETE"), headers.clone()).await;
```

Comment:
I would add check that the correct client did receive the request (not just that the response is a stolen response)